src/interpreter.js

```diff
@@ -155,57 +156,37 @@ const compileExpression = (expression) => {
 
 
 const compileLiteral = (literal) => {
-  if(literal.nodeType === 'NullLiteral') {
-    return `null`
+  const rules = {
+    NullLiteral: () => 'null',
+    SelfLiteral: () => 'this',
+    BooleanLiteral: ({ value }) => `${value}`,
+    NumberLiteral: ({ value }) => `${value}`,
+    StringLiteral: ({ value }) => `"${value}"`,
+    SetLiteral: ({ values }) => `new Set([ ${values.map(compileExpression).join()} ])`,
+    ListLiteral: ({ values }) => `[ ${values.map(compileExpression).join()} ]`
   }
-
-  if(literal.nodeType === 'SelfLiteral') {
-    return `this`
+  if (rules[literal.nodeType]) {
```

Comment:
I regret using the word "visitor", because it makes me think in Gamma's double dispatch based visitor. The version I want is pretty much just what you wrote. But I think it would be best if we join all the "rules" into a single object instead of having separate compile functions, since they all call each other anyway. Does that make sense?